lib/widgets/bottom_navigation.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:flutter/material.dart';
+import 'package:pispapp/theme/light_color.dart';
+
+class BottomNavigationButton {
+  BottomNavigationButton(this.icon, this.widget);
+
+  IconData icon;
+  Widget widget;
+}
+
+class BottomNavigation extends StatefulWidget {
+  BottomNavigation(this.buttons, this.rootWidgetChange);
+
+  final List<BottomNavigationButton> buttons;
+  final void Function(Widget) rootWidgetChange;
+
+  @override
+  _BottomNavigationState createState() => _BottomNavigationState();
+}
+
+class _BottomNavigationState extends State<BottomNavigation> {
+  int _index = 0;
+
+  BottomNavigationBarItem _icons(IconData icon) {
+    return BottomNavigationBarItem(
+      icon: Icon(
+        icon,
+      ),
+      title: const Text(
+        '',
+      ),
+    );
+  }
+
+  void _onItemTapped(int index) {
+    widget.rootWidgetChange(widget.buttons[index].widget);
```

Comment:
I think the way this works is a little bit too hacky (with passing the widget from here and making `Dashboard` a `StatefulWidget`)...

You may want to consider using `Provider` here to update the display on `Dashboard`. That way, `Dashboard` could just be a `StatelessWidget`.